decayAmplitude/massDependence.cc

```diff
@@ -564,6 +612,244 @@ piPiSWaveAuMorganPenningtonKachaev::piPiSWaveAuMorganPenningtonKachaev()
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+KPiSGLASS::KPiSGLASS(const double a, const double r, const double M0, const double G0, const double phiF, const double phiR, const double phiRsin,
+                     const double F, const double R, const double MMax, const std::string& tag)
+:
+		massDependenceImpl<KPiSGLASS>(),
+		_a(a),
+		_r(r),
+		_M0(M0),
+		_G0(G0),
+		_phiF(phiF),
+		_phiR(phiR),
+		_phiRsin(phiRsin),
+		_F(F),
+		_R(R),
+		_MMax(MMax),
+		_tag(tag),
+		_piChargedMass(0.0),
+		_kaonChargedMass(0.0)
+{
+	particleDataTable& pdt = particleDataTable::instance();
+	for (const auto& partName : { "pi+", "K+" }) {
+		if (not pdt.isInTable(partName)) {
+			printErr<< "cannot find particle " << partName << " in particle data table. " << "Aborting..." << endl;
+			throw;
+		}
+	}
+	_piChargedMass = pdt.entry("pi+")->mass();
+	_kaonChargedMass = pdt.entry("K+")->mass();
+}
+
+boost::shared_ptr<KPiSGLASS>
+KPiSGLASS::Create(const libconfig::Setting* massDepKey)
+{
+	if (massDepKey != nullptr) {
+		double a, r, M0, G0, phiF, phiR, phiRsin, F, R, MMax;
+		string tag;
+		if (not massDepKey->lookupValue("a", a)) {
```

Comment:
Not sure whether its worth the effort. But have you thought about generating the repetitive 'massDepKey->lookupValue' code using macros?

Reply:
No macros please, but reducing the duplication with a function (of whatever sort) would indeed be a good idea.